packages/dd-trace/src/plugins/util/web.js

```diff
@@ -387,7 +387,7 @@ const web = {
 
     return function (statusCode, statusMessage, headers) {
       headers = typeof statusMessage === 'string' ? headers : statusMessage
-      headers = Object.assign(res.getHeaders(), headers)
+      headers = Object.assign({}, res.getHeaders(), headers)
```

Comment:
Is it potentially so that either res.getHeaders() or headers could be empty? If that's the case, we could likely prevent the copy all together? I didn't check the overall code though.

I might be wrong but I believe this will fail our linter. Could you please check that and add an exception if that's true? I think we should add a comment about Bun one way or the other :)

Reply:
Thanks for the feedback! You're absolutely right about the optimization potential. I've implemented the change to prevent unnecessary object copying when both res.getHeaders() and headers are empty.